src/Proxy.NET/Services/ProxyService.cs

```diff
@@ -0,0 +1,111 @@
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using Proxy.NET.Models;
+
+namespace Proxy.NET.Services;
+
+public class ProxyService(IHttpClientFactory httpClientFactory, IMetricService metricService, IConfiguration configuration) : IProxyService
+{
+    private const int HttpTimeoutSeconds = 60;
+    private bool EnableUnitTesting => configuration.GetValue<bool>("EnableUnitTesting");
```

Comment:
I _get_ the point of this, but I'm not a particularly big fan of how it's done.

Maybe we should have an alternative implementation of `IProxyService` that we load if that flag is set.